File: src/test/java/aoc2015/Day06Test.java
package aoc2015;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day06Test {

    @Test
    void part1Test() {
        var day = new Day06();

        assertEquals(1000000, day.part1("turn on 0,0 through 999,999"));
        assertEquals(999000, day.part1("toggle 0,0 through 999,0"));
        assertEquals(998996, day.part1("turn off 499,499 through 500,500"));
    }

    @Test
    void part2Test() {
        var day = new Day06();

        assertEquals(1, day.part2("turn on 0,0 through 0,0"));
        assertEquals(2000001, day.part2("toggle 0,0 through 999,999"));
    }
}

File: aoc25/src/test/java/day06/Day06Test.kt
package day06

import aoc.loadExample
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class Day06Test {

    val filename = "day06.txt"

    @Test
    fun part1Example() {
        val input = loadExample(filename)
        val subject = Day06()

        val result = subject.part1(input)

        assertThat(result).isEqualTo(4277556L)
    }

    @Test
    fun part2Example() {
        val input = loadExample(filename)
        val subject = Day06()

        val result = subject.part2(input)

        assertThat(result).isEqualTo(3263827L)
    }
}
